go/vt/binlog/keyrange_filter.go

```diff
@@ -39,12 +43,26 @@ func KeyRangeFilterFunc(keyrange *topodatapb.KeyRange, sendReply sendTransaction
 						continue
 					}
 				}
-				if !key.KeyRangeContains(keyrange, keyspaceID) {
-					// Skip keyspace ids that don't belong to the destination shard.
-					continue
+				if len(keyspaceIDS) == 1 {
+					if !key.KeyRangeContains(keyrange, keyspaceIDS[0]) {
+						// Skip keyspace ids that don't belong to the destination shard.
+						continue
+					}
+					filtered = append(filtered, statement)
+					matched = true
```

Comment:
I think we can put a 'continue' here and then the next block can be un-indented, and not in the 'else'?